crates/proc-macro-srv/src/server.rs

```diff
@@ -54,6 +55,53 @@ fn spacing_to_external(spacing: Spacing) -> proc_macro::Spacing {
     }
 }
 
+fn literal_to_external(literal_kind: ast::LiteralKind) -> Option<proc_macro::bridge::LitKind> {
+    Some(match literal_kind {
+        ast::LiteralKind::String(data) => {
+            if data.is_raw() {
+                bridge::LitKind::StrRaw(data.raw_delimiter_count()?)
+            } else {
+                bridge::LitKind::Str
+            }
+        }
+        ast::LiteralKind::ByteString(data) => {
+            if data.is_raw() {
+                bridge::LitKind::ByteStrRaw(data.raw_delimiter_count()?)
+            } else {
+                bridge::LitKind::ByteStr
+            }
+        }
+        ast::LiteralKind::CString(data) => {
+            if data.is_raw() {
+                bridge::LitKind::CStrRaw(data.raw_delimiter_count()?)
+            } else {
+                bridge::LitKind::CStr
+            }
+        }
+        ast::LiteralKind::IntNumber(_) => bridge::LitKind::Integer,
+        ast::LiteralKind::FloatNumber(_) => bridge::LitKind::Float,
+        ast::LiteralKind::Char(_) => bridge::LitKind::Char,
+        ast::LiteralKind::Byte(_) => bridge::LitKind::Byte,
+        ast::LiteralKind::Bool(_) => unreachable!(),
+    })
+}
+
+fn str_to_lit_node(input: &str) -> Option<ast::Literal> {
+    let input = input.trim();
+    let source_code = format!("fn f() {{ let _ = {input}; }}");
+
+    let parse = ast::SourceFile::parse(&source_code);
```

Comment:
Thanks for the feedback, this change definitely makes better use of the API and is less of a hack.

This does raise another question for me though (apologies, I'm not familiar with the parsing APIs). I would like to use `syntax::parsing::build_tree` to obtain a `GreenNode` so I can eventually construct an AST node. Is this the correct approach? Is it ok to change `syntax::parsing::build_tree` from `pub(crate)` to `pub` and re-export it from `syntax`?

Reply:
Hmm, right. Instead of exposing that let's add a `parse` method to `ast::Literal` then, similar to `SourceFile::parse` and do the handling in the syntax crate, that should prevent us from having to expose those parts I think

Reply:
Sounds good, thank you!